db/schema_tables.cc

```diff
@@ -1327,37 +1334,38 @@ static future<> merge_tables_and_views(distributed<service::storage_proxy>& prox
     // was already dropped (see https://github.com/scylladb/scylla/issues/5614)
     auto& db = proxy.local().get_db();
     auto ts = db_clock::now();
-    co_await coroutine::parallel_for_each(views_diff.dropped, [&db, ts] (schema_diff::dropped_schema& dt) {
+    co_await max_concurrent_for_each(views_diff.dropped, max_concurrent, [&db, ts] (schema_diff::dropped_schema& dt) {
         auto& s = *dt.schema.get();
         return replica::database::drop_table_on_all_shards(db, s.ks_name(), s.cf_name());
```

Comment:
This call has a cross-shard latency, which can dominate, and we could experience regression in the speed of the drop with many tables when the system is fully loaded.

Optimal would be to pass the list of tables to the other shard and preempt there.

Reply:
I think the actual dropping will dominate, since lots of I/O is involved for removing the sstables.

Reply:
By default we take snapshots. This consumes I/O, but is probably not throughput-bound either (like we saw with the flushing of schema tables). If resources on the other shards are underutilized, then reducing parallelizm will reduce the overall speed of the operation.

Reply:
We now flush the table on all shards, then take the snapshot on all shards, then delete the sstables, on all shards in parallel, but joining the operations in between the steps.  If we working on a number of tables in parallel, using a sliding windows, why would shards be underutilized?

Reply:
For example, flushing small sstables has high latency which is not throughput-bound, due to fsyncs, meaning that you could pack more such operations without saturating the disk. We saw flushing of schema tables take 30s due to this. If this component dominates dropping, then dropping N tables takes the same amount of time as dropping 1 table, up to some N, which could be higher than 4.

Reply:
N can probably be increased to reduce the impact.